File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab2/lab2_schelet/sem1_2/model/SortingTask.java
package sem1_2.model;


import sem1_2.decorator.AbstractSorter;


import java.util.Arrays;



public class SortingTask extends Task{

    private final int[] array;
    private AbstractSorter sorter;

    public SortingTask(String id, String description, int[]array, SortingStrategy sortingStrategy) {
        super(id,description);
        this.array = array;

        switch (sortingStrategy) {
            case BUBBLE_SORTING -> this.sorter = new BubbleSort();
            case QUICK_SORTING ->  this.sorter = new QuickSort();
        }
    }

    @Override
    public void execute() {
        this.sorter.sort(this.array);
        System.out.println(Arrays.toString(array));
    }
}

File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab9/ConsoleApp1/model/SortingTask.cs
using ConsoleApp1.decorator;
namespace ConsoleApp1.model;


public class SortingTask : Task
{
    private int[] array;
    private AbstractSorter sorter ;

    public SortingTask(string id, string description, int[] array, SortingStrategy sortingStrategy) 
        : base(id, description)
    {
        this.array = array;
        this.sorter = sorter;
        switch (sortingStrategy)
        {  
          case SortingStrategy.BubbleSorting:
              this.sorter = new BubbleSort();
              break;
          case SortingStrategy.QuickSorting:
              this.sorter = new QuickSort();
              break;
        }
    }
    

    public override void Execute()
    {
        this.sorter.Sort(array);
        System.Console.WriteLine(string.Join(" ", array));
    }
}